testnet/stacks-node/src/tests/epoch_205.rs

```diff
@@ -913,3 +916,292 @@ fn test_cost_limit_switch_version205() {
+        assert_eq!(account.balance, 10492300000);
+    }
+
+    let mut ctr = 0;
+    while ctr < txs.len() {
+        submit_tx(&http_origin, &txs[ctr]);
+        if !wait_for_microblocks(&microblocks_processed, 180) {
+            // we time out if we *can't* mine any more microblocks
+            break;
+        }
+        ctr += 1;
+    }
+
+    // only one fit
+    assert_eq!(ctr, 1);
+    sleep_ms(5_000);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    eprintln!("First confirmed microblock stream!");
+
+    microblocks_processed.store(0, Ordering::SeqCst);
+
+    // send the rest of the transactions
+    while ctr < txs.len() {
+        submit_tx(&http_origin, &txs[ctr]);
+        ctr += 1;
+    }
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    eprintln!("Second confirmed microblock stream!");
+
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    // confirm it
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect epoch transition");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect microblock stream");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    // this test can sometimes miss a mine block event.
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let blocks = test_observer::get_blocks();
+    assert!(blocks.len() >= 5, "Should have produced at least 5 blocks");
+
+    let mut max_big_txs_per_microblock_20 = 0;
+    let mut total_big_txs_per_microblock_20 = 0;
+
+    let mut max_big_txs_per_microblock_205 = 0;
+    let mut total_big_txs_per_microblock_205 = 0;
+
+    let mut in_205 = false;
+
+    // NOTE: this only counts the number of txs per stream, not in each microblock
+    for block in blocks {
+        let transactions = block.get("transactions").unwrap().as_array().unwrap();
+        eprintln!("{}", transactions.len());
+
+        let mut num_big_microblock_txs = 0;
+
+        for tx in transactions.iter() {
+            let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+            if raw_tx == "0x00" {
+                continue;
+            }
+            let tx_bytes = hex_bytes(&raw_tx[2..]).unwrap();
+            let parsed = StacksTransaction::consensus_deserialize(&mut &tx_bytes[..]).unwrap();
+            if let TransactionPayload::SmartContract(tsc) = parsed.payload {
+                if tsc.name.to_string().find("costs-2").is_some() {
+                    in_205 = true;
+                }
+                if tsc.name.to_string().find("large").is_some() {
+                    if in_205 {
+                        num_big_microblock_txs += 1;
```

Comment:
move this line outside if it's in both if/else

Reply:
Done